plugins/inputs/logfile/tail/tail.go

```diff
@@ -225,6 +226,16 @@ func (tail *Tail) readLine() (string, error) {
 	tail.lk.Lock()
 	defer tail.lk.Unlock()
 
+	// do not allow line reading if the output pusher size is greater than configured
+	block, bufferSize, maxBufferSize := util.GetLogBlocker().Block()
+	for block {
+		tail.Logger.Infof("max buffer of logs size sending to cloudwatch " +
+			"blocking reading for one second for file %s max buffer %d current buffer %d",
+			tail.Filename, maxBufferSize, bufferSize)
+		time.Sleep(time.Second)
```

Comment:
Why do we need to wait a whole second?

Reply:
No. This can change to a different period of time if you have any ideas.